lambda/service/store/store.go

```diff
@@ -72,3 +78,108 @@ func (r *ApplicationDatabaseStore) Delete(ctx context.Context, applicationId int
+	query := fmt.Sprintf("SELECT dataset_id, user_id, role from \"%[1]v\".dataset_user where dataset_id=%[2]v and user_id in (SELECT integration_user_id from \"%[1]v\".webhooks where id=$1)", r.OrganizationID, datasetId)
+	queryContext, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+	var dataserUser DatasetUser
+	err := r.DB.QueryRowContext(queryContext, query, applicationId).Scan(
+		&dataserUser.DatasetID,
+		&dataserUser.UserID,
+		&dataserUser.Role)
+	if err != nil {
+		return nil, err
+	}
+
+	return &dataserUser, nil
+}
+
+func (r *ApplicationDatabaseStore) GetDatasetUserByUserId(ctx context.Context, userId int64, datasetId int64) (*DatasetUser, error) {
+	query := fmt.Sprintf("SELECT dataset_id, user_id, role from \"%[1]v\".dataset_user where dataset_id=%[2]v and user_id=$1", r.OrganizationID, datasetId)
+	queryContext, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+	var dataserUser DatasetUser
+	err := r.DB.QueryRowContext(queryContext, query, userId).Scan(
+		&dataserUser.DatasetID,
+		&dataserUser.UserID,
+		&dataserUser.Role)
+	if err != nil {
+		return nil, err
+	}
+
+	return &dataserUser, nil
+}
+
+// utility store methods
+func (r *ApplicationDatabaseStore) InsertOrganizationUser(ctx context.Context, organizationUser OrganizationUser) (int64, error) {
+	var organization_id int64
+	query := "insert into pennsieve.organization_user (organization_id,user_id,permission_bit) VALUES ($1,$2,$3) RETURNING ORGANIZATION_ID"
+	queryContext, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+	if err := r.DB.QueryRowContext(
+		queryContext,
+		query,
+		organizationUser.OrganizationID,
+		organizationUser.UserID,
+		organizationUser.PermissionBit).Scan(&organization_id); err != nil {
+		return 0, err
+	}
+	return organization_id, nil
+}
+
+func (r *ApplicationDatabaseStore) DeleteOrganizationUser(ctx context.Context, organization_id int64, user_id int64) error {
+	query := "DELETE from pennsieve.organization_user WHERE organization_id=$1 and user_id=$2"
+	queryContext, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+	err := r.DB.QueryRowContext(queryContext, query, organization_id, user_id)
+	if err != nil {
+		return err.Err()
+	}
+	return nil
+}
+
+func (r *ApplicationDatabaseStore) InsertDatasetUser(ctx context.Context, datasetUser DatasetUser) (int64, error) {
+	var dataset_id int64
+	query := fmt.Sprintf("insert into \"%v\".dataset_user (dataset_id,user_id,role) VALUES ($1,$2,$3) RETURNING DATASET_ID",
+		r.OrganizationID)
+	queryContext, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+	if err := r.DB.QueryRowContext(
+		queryContext,
+		query,
+		datasetUser.DatasetID,
+		datasetUser.UserID,
+		datasetUser.Role).Scan(&dataset_id); err != nil {
+		return 0, err
+	}
+	return dataset_id, nil
+}
+
+func (r *ApplicationDatabaseStore) DeleteDatasetUser(ctx context.Context, dataset_id int64, user_id int64) error {
```

Comment:
Why would the Authorizer be able to create and delete users in the platform? Just for the tests? Does it make sense to separate that somehow?

Reply:
@jwagenaar these are just utility functions for the tests. I can perhaps separate this so it is clearer.